xray/segment.go

```diff
@@ -53,7 +54,7 @@ func BeginFacadeSegment(ctx context.Context, name string, h *header.Header) (con
 }
 
 // BeginSegment creates a Segment for a given name and context.
-func BeginSegment(ctx context.Context, name string) (context.Context, *Segment) {
+func BeginSegment(ctx context.Context, name string, sample bool) (context.Context, *Segment) {
```

Comment:
As discussed, we do not want to break the external API here. We should probably add a new API, e.g. `BeginSegmentWithSampling`, that has these parameters.

Reply:
Right. Makes Sense!